crates/universal-wallet/fuzz/src/lib.rs

```diff
@@ -0,0 +1,128 @@
+use arbitrary::{Arbitrary, Unstructured};
+use borsh::{BorshDeserialize, BorshSerialize};
+use sov_modules_api::macros::UniversalWallet;
+use sov_modules_api::prelude::serde::{Deserialize, Serialize};
+use sov_modules_api::SafeString;
+use sov_universal_wallet::schema::safe_string::DEFAULT_MAX_STRING_LENGTH;
+
+mod js_compat;
+
+#[cfg(feature = "js-compat")]
+pub mod types {
+    pub type I64 = crate::js_compat::JsI64;
+    pub type I128 = crate::js_compat::JsI128;
+    pub type U64 = crate::js_compat::JsU64;
+    pub type U128 = crate::js_compat::JsU128;
+}
+
+#[cfg(not(feature = "js-compat"))]
+pub mod types {
+    pub type I64 = i64;
+    pub type I128 = i128;
+    pub type U64 = u64;
+    pub type U128 = u128;
+}
+
+use types::{I128, I64, U128, U64};
+
+// arbitrary isn't implemented for safe string
+#[derive(Debug, BorshSerialize, BorshDeserialize, Serialize, Deserialize, UniversalWallet)]
+pub struct ArbitrarySafeString(SafeString);
+
+impl Arbitrary<'_> for ArbitrarySafeString {
+    fn arbitrary(u: &mut Unstructured<'_>) -> arbitrary::Result<Self> {
+        let len = u.int_in_range(0..=DEFAULT_MAX_STRING_LENGTH)?;
+
+        let chars: Result<String, _> = (0..len)
+            .map(|_| {
+                let c = u.int_in_range(32u8..=126u8)? as char;
+                Ok(c)
+            })
+            .collect();
+
+        let s = chars?;
+        Ok(ArbitrarySafeString(s.try_into().unwrap()))
+    }
+}
+
+#[derive(
+    Debug, BorshSerialize, BorshDeserialize, Serialize, Deserialize, UniversalWallet, Arbitrary,
+)]
+pub enum ByteVecInput {
+    Hex(#[sov_wallet(display(hex))] Vec<u8>),
+    Base58(#[sov_wallet(display(base58))] Vec<u8>),
+    Decimal(#[sov_wallet(display(decimal))] Vec<u8>),
+}
+
+#[derive(
+    Debug, BorshSerialize, BorshDeserialize, Serialize, Deserialize, UniversalWallet, Arbitrary,
+)]
+pub enum ByteArrayInput {
+    Hex(#[sov_wallet(display(hex))] [u8; 32]),
+    Base58(#[sov_wallet(display(base58))] [u8; 32]),
+    Decimal(#[sov_wallet(display(decimal))] [u8; 32]),
+}
+
+#[derive(
+    Debug, BorshSerialize, BorshDeserialize, Serialize, Deserialize, UniversalWallet, Arbitrary,
+)]
+pub enum NumberInput {
+    U8(u8),
+    U16(u16),
+    U32(u32),
+    U64(U64),
+    U128(U128),
+    I8(i8),
+    I16(i16),
+    I32(i32),
+    I64(I64),
+    I128(I128),
+    #[cfg(feature = "floats")]
```

Comment:
https://github.com/Sovereign-Labs/sovereign-sdk/issues/1451

Need to fix floats, not needed right now since with pure JS impl we won't need json to borsh